### ISSUE TITLE
No Support Yet For Opening And/Or Editing HamX Image Files

### ISSUE THREAD

**Issue by blenderbach** [photopea/photopea#5139]
Back in the 90s and very early 2000s, Blender had an obscure image format called "HamX", that it used for preview purposes. This is a compact 8 bits RLE compressed format. When I put the file to be analyzed for the format, the online file analyzer, said it was a "Generic IFF container". Therefore, I am uploading a HamX image file here, for analysis, and maybe Photopea can be the only software that can actually edit and use the HamX format!

P.S. I don't know the extension of HamX files, as they don't have one when being exported. Blender just knows that it's a HamX file. I used Blender 1.60 to render this HamX image file. Since I don't have an extension, I just added ".hamx" as a placeholder extension.
[planet0058.zip](https://github.com/photopea/photopea/files/10120887/planet0058.zip)

**Comment by photopea** [comment]
Photopea can load "something" out of your file now.

Do you maybe have more files? Or even better, a screenshot of what the file should look like?

**Comment by blenderbach** [comment]
> Photopea can load "something" out of your file now.
> 
> Do you maybe have more files? Or even better, a screenshot of what the file should look like?

I am impressed. REALLY impressed! The image isn't really correct, BUT it is actually resembling it somewhat, which means that the "decoding" is almost there! :) Of course it's not correct by any means, and it's duplicated 4 times, and it looks completely wrong, but you can actually see something that resembles the original image! I have a JPG version of the HamX here, compared to the HamX decoding on Photopea, so this might help with further "wizardry". Good Luck! 👍  

_**The Original Image**_
![planet0058](https://user-images.githubusercontent.com/87039308/205132257-558f87fd-edb2-4f77-bcda-17f08cb27e65.jpg)

_**The Photopea Decoding**_
![image](https://user-images.githubusercontent.com/87039308/205132135-0a2b4527-f69e-4c98-b57c-4b73f42f8d66.png)

**Comment by photopea** [comment]
Are you sure that the HAMX file contains a colored version of your JPG? Or maybe it contains just a grayscale version?

**Comment by blenderbach** [comment]
> Are you sure that the HAMX file contains a colored version of your JPG? Or maybe it contains just a grayscale version?

Yep. :) Here's a screenshot when I opened the HamX image in Blender 1.60.
![image](https://user-images.githubusercontent.com/87039308/205138515-3aca6db8-4391-442e-a935-3dc403c9d6bc.png)

**Comment by photopea** [comment]
Ok, it should work in Photopea now :) please, check if your other images also work.

**Comment by blenderbach** [comment]
> Ok, it should work in Photopea now :) please, check if your other images also work.

Yeowza wowza! That was quick! I don't know how you manage to get these formats implemented so fast, but you are one heck of a programming wizard! Thanks, and Good Job!👏👏👏 I'm definitely going to have to tell the entire Blender Community about this! This is an amazing step in history! HamX files are finally possible to be opened! :) Amazing!

**Comment by photopea** [comment]
You are welcome :)
